### DISCUSSION TITLE
Metrics for OPA evaluation timeouts?

### DISCUSSION THREAD

**Discussion by ghost** [open-policy-agent/community#208] [Envoy] [answered]
Does the OPA sidecar emit any metrics corresponding to evaluation timeouts?
When trying to debug scenarios like the one discussed [here](https://github.com/open-policy-agent/opa/issues/3076#issuecomment-763572986), found that metrics like builtin http latency and OPA evaluation latency are not emitted when the query execution ends abruptly. There is just an error with message `context deadline reached during query execution`, but not much info to confirm an evaluation timeout. A dedicated metric can help. Thanks!

PS: Alternatively, if it can be GUARANTEED that all scenarios with error message `context deadline reached during query execution` are due to evaluation timeouts only, the OPA decision logs might serve the purpose.

**Comment by anderseknert** [comment, marked as answer]
Hey there! Is that deadline not in the calling service, i.e. envoy in that case? Or is that returned by OPA?

**Reply by ghost** [reply]
The deadline is set at Envoy, but this error message `context deadline reached during query execution` is part of the decision logs emitted by OPA.

**Reply by anderseknert** [reply]
Alright. Not sure about guarantees, but FWIW I've never seen that error message in any other context :)

**Reply by ghost** [reply]
In that case, should OPA emit dedicated metrics for eval timeouts? Shouldn't be an intrusive change AFAICT, what's your opinion?
